samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java

```diff
@@ -594,7 +593,13 @@ public void deleteMessages(Map<SystemStreamPartition, String> offsets) {
           .collect(Collectors.toMap(entry ->
               new TopicPartition(entry.getKey().getStream(), entry.getKey().getPartition().getPartitionId()),
               entry -> RecordsToDelete.beforeOffset(Long.parseLong(entry.getValue()) + 1)));
-      adminClient.deleteRecords(recordsToDelete);
+
+      adminClient.deleteRecords(recordsToDelete).all().whenComplete((ignored, exception) -> {
+        if (exception != null) {
+          LOG.error("Delete message failed for SSPs {} due to {}", offsets.keySet(), exception);
```

Comment:
I think this will end up not printing out the stack trace. I think you need to use the `LOG.error(String, Exception)` instead.

Reply:
1. Since this is an optimization in the background to clean up data for intermediate topics, it is not be desirable to bubble it up and kill the application.
2. The return future is of type Void and ignored doesn't carry any information.